Tests/Unit/Hooks/HookProviderTest.php

```diff
@@ -0,0 +1,365 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OliverKlee\Seminars\Tests\Unit\Hooks;
+
+use Nimut\TestingFramework\TestCase\UnitTestCase;
+use OliverKlee\Seminars\Hooks\HookProvider;
+use OliverKlee\Seminars\Hooks\Interfaces\Hook;
+use OliverKlee\Seminars\Tests\Unit\Fixtures\Hooks\TestingHookInterface;
+use OliverKlee\Seminars\Tests\Unit\Fixtures\Hooks\TestingHookInterfaceNotExtendsHook;
+use OliverKlee\Seminars\Tests\Unit\Fixtures\Hooks\TestingHookImplementor;
+use OliverKlee\Seminars\Tests\Unit\Fixtures\Hooks\TestingHookImplementor2;
+use OliverKlee\Seminars\Tests\Unit\Fixtures\Hooks\TestingHookImplementorNotImplementsInterface;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+/**
+ * Test case.
+ *
+ * @author Michael Kramer <m.kramer@mxp.de>
+ */
+class HookProviderTest extends UnitTestCase
+{
+    /**
+     * @var array
+     */
+    private $extConfBackup = [];
+
+    protected function setUp()
+    {
+        $this->extConfBackup = $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'];
```

Comment:
Oh, and for installations that already have some hooks provided, please unset the array here.

We also should have a test for cases that the element not being set as well as an empty array does not cause problems.

Reply:
> Oh, and for installations that already have some hooks provided, please unset the array here.
I don't understand that part. Installations of what?

Reply:
TYPO3 installations of the seminars extension that run the tests.